### DISCUSSION TITLE
Welcome to evrete Discussions!

### DISCUSSION THREAD

**Discussion by andbi** [evrete/evrete#8] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by divya544** [comment]
Can we able to write rules in excel and fetch the rules in our code using evrete rule engine?

**Reply by andbi** [reply]
@divya544  of course! 

Personally, I use them wherever possible. However, the problem with Excel sheets is that if you want them to be user-friendly, they should always be domain-specific. There's no general Excel rule format that would fit, for example, both insurance- and pharmacy-related tasks. 

Here's what I do if I want to create an Excel-based rule parser:

- Design the future Excel rule format - how rules are separated, where the engine can find fact declarations, conditions, and actions.
- Include the https://poi.apache.org library for parsing Excel documents.
- Create an Excel scanner that would read provided documents according to the agreed format and use the `RuleBuilder` interface to create rules.

**Reply by divya544** [reply]
Thank you for helping. But I don't want to use apache poi.
